spec/mastermind_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'mastermind'
+
+
+game = Mastermind.new
+
+RSpec.describe Mastermind, "Print Directions" do
+
+	it "prints game play instructions to console" do
+		expect {game.print_directions}.to output(/To play:/).to_stdout 
+	end
+
+end
+
+RSpec.describe SecretCode, "Generate" do 
+	game = Mastermind.new
+	it "generates a secret code that only includes valid colors" do
+		secret_code = SecretCode.new
+		code = secret_code.generate(game.colors)
+		colors = game.colors
+		pass = true
+
+		code.each do |element| 
+			if (!(colors.include?(element)))
+				pass = false
+			end
+		end
+
+		expect(pass).to eq(true)
```

Comment:
To make it a little bit easier on yourself, you can avoid having to add that "business logic" to test if the code elements exist in the color collection by using the [include](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/include-matcher) matcher.

you could call

```rb
expect(colors).to include(*code);
```